packages/block-library/src/search/index.php

```diff
@@ -139,11 +174,42 @@ function render_block_core_search( $attributes, $content, $block ) {
 		if ( $button->next_tag() ) {
 			$button->add_class( implode( ' ', $button_classes ) );
 			if ( 'expand-searchfield' === $attributes['buttonBehavior'] && 'button-only' === $attributes['buttonPosition'] ) {
-				$button->set_attribute( 'aria-label', __( 'Expand search field' ) );
-				$button->set_attribute( 'data-toggled-aria-label', __( 'Submit Search' ) );
-				$button->set_attribute( 'aria-controls', 'wp-block-search__input-' . $input_id );
-				$button->set_attribute( 'aria-expanded', 'false' );
-				$button->set_attribute( 'type', 'button' ); // Will be set to submit after clicking.
+				if ( gutenberg_should_block_use_interactivity_api( 'core/search' ) ) {
+					$aria_label_expanded  = __( 'Submit Search' );
+					$aria_label_collapsed = __( 'Expand search field' );
+					wp_store(
+						array(
+							'state'     => array(
+								'core' => array(
+									'search' => array(
+										'ariaLabelCollapsed' => $aria_label_collapsed,
+										'ariaLabelExpanded'  => $aria_label_expanded,
+									),
+								),
+							),
+							'selectors' => array(
+								'core' => array(
+									'search' => array(
+										'ariaLabel'    => 'true' === $open_by_default ? $aria_label_expanded : $aria_label_collapsed,
+										'ariaControls' => 'true' === $open_by_default ? null : $input_id,
+										'type'         => 'true' === $open_by_default ? 'submit' : 'button',
```

Comment:
One thing's not clear to me: 

If `$open_by_default` is hardcoded to be `false` on [this line](https://github.com/WordPress/gutenberg/pull/53343/files#diff-ef7ce82e804956a7d5c001fefbceabf1820872331c6282c9d359cac1e7390734R67) and never changed as far as I can see, then why do we still need those conditionals?

Reply:
I added those conditionals so they follow the same logic we have in the client. I thought that this way, if for any reason we decide to change the `$open_by_default` in some scenarios, it is clear which attributes would be affected.

Reply:
Oh, got it. For me, it was slightly confusing because the existence of the `$open_by_default` variable implies that there are cases when it's **not** open by default. Just a comment above [this line](https://github.com/WordPress/gutenberg/pull/53343/files#diff-ef7ce82e804956a7d5c001fefbceabf1820872331c6282c9d359cac1e7390734R67) mentioning that, for now, `open_by_default` is a constant should be enough to clarify that.